includes/class-sensei-guest-user.php

```diff
@@ -0,0 +1,89 @@
+<?php
+/**
+ * Guest User
+ *
+ * Handles operations related to allowing guest users take a course.
+ *
+ * @package Sensei\Frontend
+ * @since 1.3.0
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Sensei Guest User Class.
+ *
+ * @author Automattic
+ *
+ * @since $$next-version$$
+ * @package Core
+ */
+class Sensei_Guest_User {
+
+	/**
+	 * Keeps a reference to the Guest Student role object
+	 *
+	 * @access protected
+	 * @since $$next-version$$
+	 *
+	 * @var string
+	 */
+	protected $guest_student_role = 'guest_student';
+
+	/**
+	 * Sensei_Guest_User constructor.
+	 *
+	 * @since $$next-version$$
+	 */
+	public function __construct() {
+
+		// Check if the Guest Student role exists.
+		$guest_role = get_role( $this->guest_student_role );
+
+		// If Guest Student is not a valid WordPress role create it.
+		if ( ! is_a( $guest_role, 'WP_Role' ) ) {
+			// Create the role.
+			add_role( $this->guest_student_role, __( 'Guest Student', 'sensei-lms' ) );
+		}
```

Comment:
Looks like role setup is [done in the activation hook](https://github.com/Automattic/sensei/blob/c2dcf2045e5b0f2bbdf29c6f68d3fd6df70987a9/includes/class-sensei.php#L1444) for the other parts of the plugin, should we do this one from there too? 
Even if not, for better readability this could move to a `create_role()` method.

Reply:
Thanks for the nice suggestion! It actually would be the nicest place to put the role setup code in, but it seems that portion only runs during activation. So I was thinking in case it doesn't run from existing plugin users.

I moved the role creation to separate method here https://github.com/Automattic/sensei/pull/6259/commits/d7493e63d3b672bea69dee25cc7a296e9e013f31 . Thanks!

Reply:
Right, I wasn't sure if that hook would also run after a plugin update — sadly it doesn't.